### DISCUSSION TITLE
Problem with p12 with 2 key rsa, 2 certificates and two root certificates

### DISCUSSION THREAD

**Discussion by dxeox** [MatthiasValvekens/pyHanko#455] [Q&A]
When I use a signature with a root certificate, an intermediate certificate, an RSA key, and the certificate, I have no problem signing from a p12 file; that is, I obtain a valid signature. However, when I attempt to use a signature with 2 RSA keys, 2 certificates, 1 intermediate certificate, and 2 root certificates, I encounter an issue where the signature process completes successfully, but the signature is invalid.

Here is the screenshot of the certificate causing the issue:
![cert](https://github.com/user-attachments/assets/3bd0ce5f-dbeb-4e14-9a61-7bbf565dcd5e)

**Comment by MatthiasValvekens** [comment]
Hi,

This could be because of any number of reasons that are hard/impossible to diagnose in the abstract. If you can share some relevant code and/or output documents, that'd help.
